creed/tiling_utils_test.py

```diff
@@ -350,180 +357,155 @@ def test_generate_tiled_region_fov_list(randomize_setting, moly_region,
-                # as its sorted version
-                misc_utils.verify_same_elements(
-                    computed_center_points=center_points[fov_1_end:],
-                    actual_center_points=actual_center_points_sorted[fov_1_end:]
-                )
-
-                # however, fov 2 sorted entries should NOT equal fov 2 random entries
-                # NOTE: due to randomization, this test will fail once in a blue moon
-                assert center_points[fov_1_end:] != actual_center_points_sorted[fov_1_end:]
-            # both regions are randomized
-            else:
-                # ensure the random center points for fov 1 contain the same elements
-                # as its sorted version
-                misc_utils.verify_same_elements(
-                    computed_center_points=center_points[:fov_1_end],
-                    actual_center_points=actual_center_points_sorted[:fov_1_end]
-                )
-
-                # however, fov 1 sorted entries should NOT equal fov 1 random entries
-                # NOTE: due to randomization, this test will fail once in a blue moon
-                assert center_points[:fov_1_end] != actual_center_points_sorted[:fov_1_end]
-
-                # ensure the random center points for fov 2 contain the same elements
-                # as its sorted version
-                misc_utils.verify_same_elements(
-                    computed_center_points=center_points[fov_1_end:],
-                    actual_center_points=actual_center_points_sorted[fov_1_end:]
-                )
-
-                # however, fov 2 sorted entries should NOT equal fov 2 random entries
-                # NOTE: due to randomization, this test will fail once in a blue moon
-                assert center_points[fov_1_end:] != actual_center_points_sorted[fov_1_end:]
-
-
-def test_generate_tma_fov_list():
-    # file path validation
-    with pytest.raises(FileNotFoundError):
-        tiling_utils.generate_tma_fov_list(
-            'bad_path.json', 3, 3
-        )
+        # if only the second fov is randomized
+        elif randomize_setting == ['N', 'Y']:
+            # ensure the fov 1 center points are the same for both sorted and random
+            assert center_points[:fov_1_end_pos] == actual_center_points_sorted[:fov_1_end_pos]
+
+            # ensure the random center points for fov 2 contain the same elements
+            # as its sorted version
+            misc_utils.verify_same_elements(
+                computed_center_points=center_points[fov_1_end_pos:],
+                actual_center_points=actual_center_points_sorted[fov_1_end_pos:]
+            )
+
+            # however, fov 2 sorted entries should NOT equal fov 2 random entries
+            # NOTE: due to randomization, this test will fail once in a blue moon
+            assert center_points[fov_1_end_pos:] != actual_center_points_sorted[fov_1_end_pos:]
```

Comment:
the if/elif/elif branches here can be condensed as follows:
```
order = [1, -1]
slice_args = (None, fov_1_end_pos)
for fov_idx, setting in enumerate(randomize_setting):
    slice_obj = slice(*slice_args[slice(None, None, order[fov_idx])])
    if setting == 'N':
        assert center_points[slice_obj] == acutal_center_points_sorted[slice_obj]
    else:
        misc_utils.verify_same_elements(
            computed_center_points=center_points[slice_obj],
            actual_center_points=actual_center_points_sorted[slice_obj]
        )
        assert center_points[slice_obj] != actual_center_points_sorted[slice_obj]
```

Reply:
For this particular section I like explicitly separating out the different randomize cases `['N', 'N'], ['N', 'Y'], ['Y', 'Y']`; makes it a bit easier to follow along. Thoughts @ngreenwald?

Reply:
Yeah I agree